ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/TeamsIntegrationForm.tsx

```diff
@@ -41,9 +42,7 @@ function TeamsIntegrationForm({
     initialValues = null,
     isEditable = false,
 }: IntegrationFormProps<TeamsIntegration>): ReactElement {
-    const formInitialValues = initialValues
-        ? { ...defaultValues, ...initialValues }
-        : defaultValues;
+    const formInitialValues = merge({}, defaultValues, initialValues) as TeamsIntegration;
```

Comment:
Stephan, slight nuance of TypeScript:
1. `as TeamsIntegration`
    > Sometimes you will have information about the type of a value that TypeScript can’t know about.
    > In this situation, you can use a *type assertion* to specify a more specific type:
    https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions
2. `const formInitialValues: TeamsIntegration = …;`
    > You can optionally add a type annotation to explicitly specify the type of the variable.
    > Wherever possible, TypeScript tries to automatically infer the types in your code. 
    https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-annotations-on-variables

Although TypeScript seems to infer the type here, explicit **annotation** (option 2) is type safety in case of future code change that breaks the assumption.

I tested the temporary change locally with `npm run tsc` and `npm run lint` in ui folder

Reply:
Changed to option 2.